.github/workflows/python_build_modified_version_tests.yml

```diff
@@ -0,0 +1,112 @@
+name: python_build_modified_version_tests
+on: [pull_request, push]
+jobs:
+  setup_tests:
+    runs-on: ubuntu-latest
+    outputs:
+      versions: ${{steps.modified-versions.outputs.versions}}
+    steps:
+      - uses: actions/checkout@v2
+      # Normally, we would use the superbly maintained...
+      # - uses: actions/setup-python@v2
+      #   with:
+      #    python-version: ${{ matrix.python-version }}
+      # ... but in the repo, we want to test pyenv builds on Ubuntu
+      - run: pwd
+      - env:
+          PYENV_ROOT: /home/runner/work/pyenv/pyenv
+        run: |
+          echo $PYENV_ROOT
+          echo "$PYENV_ROOT/shims:$PYENV_ROOT/bin" >> $GITHUB_PATH
+      - run: git fetch origin
+      - id: modified-versions
+        run: |
+          versions=`git diff --name-only origin/master \
+          | grep '^plugins/python-build/share/python-build/' \
+          | awk -F '/' '{print $NF}' \
+          | xargs`
+          echo $versions
+          echo "::set-output name=versions::$versions"
+
+  setup_matrix:
+    needs: setup_tests
+    if: needs.setup_tests.outputs.versions != ''
+    runs-on: ubuntu-latest
+    outputs:
+      versions: ${{steps.version_matrix.outputs.versions}}
+    steps:
+      - id: version_matrix
+        env: 
+          VERSIONS: ${{needs.setup_tests.outputs.versions}}
+        run: echo "::set-output name=versions::`echo $VERSIONS | xargs -n1 | jq -R . | jq -sc .`"
```

Comment:
Could this be combined into `setup_tests`? I can't readily see why it's necessary as a separate step.